addons/account/report/account_invoice_report.py

```diff
@@ -107,7 +107,7 @@ def _select(self):
                 line.quantity / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0) * (CASE WHEN move.type IN ('in_invoice','out_refund','in_receipt') THEN -1 ELSE 1 END)
                                                                             AS quantity,
                 -line.balance                                               AS price_subtotal,
-                -line.balance / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0)
+                -line.balance / NULLIF(line.quantity / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0), 0.0)
```

Comment:
This is wrong: division by zero.
`-COALESCE(line.balance / NULLIF(line.quantity, 0.0) / NULLIF(COALESCE(uom_line.factor, 1), 0.0) / NULLIF(COALESCE(uom_template.factor, 1), 0.0), 0.0)`

For readability:
```
`-COALESCE(line.balance 
    / NULLIF(line.quantity, 0.0) 
    / NULLIF(COALESCE(uom_line.factor, 1), 0.0) 
    / NULLIF(COALESCE(uom_template.factor, 1), 0.0),
0.0)`
```

Reply:
Weird, I tested this query and it was NULL when the denominator in 0, but better safe than sorry, I've updated the code with your suggestion